t04_refactored.py

```diff
@@ -161,11 +161,98 @@ def team_6_adv(username, delay):
+    print("Staying here is certainly not wise. You must find your way out.")
+    print("\n")
+    OurSleep()
+
+    direction = input("Which direction would you like to go? [North/South/East/West]")
+
+    if direction == "North":
+        # Good choice!
+        print("You are still trapped in the dark, but someone else is there with you now! I hope they're friendly...")
+        OurSleep()
+    elif direction == "South":
+        # Oh... Bad choice
+        print("You hear a growl. Not a stomach growl. More like a big nasty animal growl.")
+        OurSleep()
+        print("Oops. Turns out the cave was home to a nasty grizzly bear. ")
+        print("Running seems like a good idea now. But... it's really, really dark.")
+        print("You turn and run like hell. The bear wakes up to the sound of your head bouncing off a low stalactite. ")
+        print("He eats you. You are delicious.")
+        dead = True
+    else:
+        # Neutral choice
+        print(
+            "You're in another part of the cave. It is equally dark, and equally uninteresting. Please get me out of here!")
+        OurSleep()
+
+    if dead:
+        print("Oh no! You died. Better luck next time! Try again by hitting the green play button. ")
+        quit()
+
+    bridge = input("You come across three bridges do you want to take the [Left/Middle/Right] path?")
+
+    if bridge == "Left":
+        print("You walk across a stone pathway confronted with a beautiful waterfall at the end: [+5 inspiration]")
+        OurSleep()
+
+    # if they go left: You come across a beautiful waterfall: [+5 inspiration]
+
+    elif bridge == "Middle":
+        print("You walk across a chasm and the wooden bridge breaks and you are about to fall: [THINK FAST]")
+        OurSleep()
+
+        rope = float(input(
+            "Quick! Catch yourself using a rope! Determine its length in feet by picking a number between [1] to [20]"))
+
+        if rope > 13:
+            print(
+                "Phew! You managed to grab the rope just in time before the fall! Congratulations: [+5 Survival Skills]")
+        else:
+            print("Oh no! Your hand slips and you fal backwards into the chasm: [DEAD]")
+            dead = True
+
+
+    # elif they go middle: You walk across a chasm and the wooden bridge breaks and you fall: [DEAD]
+
+    else:
+        print("You walk safely across a plain bridge: [0 inspiration]")
+        sleep(delay)
+    # else they go right: You walk safely across a plain bridge: [0 inspiration]
+
+    if dead:
+        print("Oh no! You did not make it to the end: [GAME OVER]")
+        return False
+
+
+    # The following is the end of the story. Don't change this section, unless you really want to.
+    print("Look at that! You made it to the end of the story without dying! ")
+    print("Congratulations... now go play again and find an interesting way to perish. ")
+    print("Try again by hitting the green play button.")
+
```

Comment:
Check out main... it tries to kill the user at the end of every function. Your function ends to return True (alive) or False (dead). See lines 398 and 399